src/v3/src/components/Icon/OdysseyIcons.tsx

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright (c) 2023-present, Okta, Inc. and/or its affiliates. All rights reserved.
+ * The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
+ *
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+import type { SvgIconProps } from '@mui/material/SvgIcon';
+import {
+  AppsIcon as _AppsIcon,
+  ArrowLeftIcon as _ArrowLeftIcon,
+  ArrowRightIcon as _ArrowRightIcon,
+  CheckCircleFilledIcon as _CheckCircleFilledIcon,
+  CloseIcon as _CloseIcon,
+  DevicesIcon as _DevicesIcon,
+  GlobeIcon as _GlobeIcon,
+  HideIcon as _HideIcon,
+  InformationCircleIcon as _InformationCircleIcon,
+  ShowIcon as _ShowIcon,
+  UserIcon as _UserIcon,
+} from '@okta/odyssey-react-mui/icons';
+import type { FunctionComponent, Ref } from 'preact';
+
+export interface SvgIconNoChildrenProps extends Omit<SvgIconProps, 'children'> {
+  /**
+   * Children are never rendered.
+   */
+  children?: never;
+}
+
+export type OdysseyIcon = FunctionComponent<Omit<SvgIconNoChildrenProps, 'ref'> & { ref?: Ref<SVGSVGElement> }>;
+
+// Export all icons as of type OdysseyIcon
+export const AppsIcon = _AppsIcon as OdysseyIcon;
+export const ArrowLeftIcon = _ArrowLeftIcon as OdysseyIcon;
+export const ArrowRightIcon = _ArrowRightIcon as OdysseyIcon;
+export const CheckCircleFilledIcon = _CheckCircleFilledIcon as OdysseyIcon;
+export const CloseIcon = _CloseIcon as OdysseyIcon;
+export const DevicesIcon = _DevicesIcon as OdysseyIcon;
+export const GlobeIcon = _GlobeIcon as OdysseyIcon;
+export const HideIcon = _HideIcon as OdysseyIcon;
+export const InformationCircleIcon = _InformationCircleIcon as OdysseyIcon;
+export const ShowIcon = _ShowIcon as OdysseyIcon;
+export const UserIcon = _UserIcon as OdysseyIcon;
```

Comment:
Can you explain how this type conversion works? I was also trying to figure out why TS was letting us directly render icons imported from `@okta/odyssey-react-mui/icons`